### DISCUSSION TITLE
make a shutdown script blok

### DISCUSSION THREAD

**Discussion by theTWIXhunter** [ScratchAddons/ScratchAddons#6982] [Ideas] [answered]
create a button on the title bar that sends a signal to the project that it should stop, so that the project can reset things (e.g. for multiplayer) and that it's not an immediate stop (a bit similar to the close program button (windows) and killing a project (task manager)

something like:

when
stop
clicked

**Comment by theTWIXhunter** [comment]
https://scratch.mit.edu/discuss/topic/731234/?page=1#post-7710438

**Comment by CubesterYT** [comment]
Scratch Addons can't create new blocks, as that needs to be made by Scratch themselves by adding it to their VM. Scratch Addons is just a browser extension that modifies the website, we can't modify the VM.

**Comment by Samq64** [comment, marked as answer]
There's a workaround for that: https://en.scratch-wiki.info/wiki/Stop_Detection
